src/test/java/org/elasticsearch/cluster/metadata/MetaDataTests.java

```diff
@@ -579,76 +592,157 @@ public void concreteIndicesWildcardExpansion() {
+            case 2:
+                allIndices = new String[] { MetaData.ALL };
+                break;
+            }
+
+            IndicesOptions indicesOptions = IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean());
+            MetaData metadata = MetaData.builder().build();
+
+            // with no indices, asking for all indices should return empty list or exception, depending on indices options
+            if (indicesOptions.allowNoIndices()) {
+                String[] concreteIndices = metadata.concreteIndices(indicesOptions, allIndices);
+                assertThat(concreteIndices, notNullValue());
+                assertThat(concreteIndices.length, equalTo(0));
+            } else {
+                try {
+                    metadata.concreteIndices(indicesOptions, allIndices);
+                    fail("calling concreteIndices for _all indices with allowNoIndices=false should throw exception");
+                } catch (ElasticsearchIllegalArgumentException | IndexMissingException e) {
+                    // expected exception
+                }
+            }
+
+            // with existing indices, asking for all indices should return all open/closed indices depending on options
+            metadata = MetaData.builder()
+                    .put(indexBuilder("aaa").state(State.OPEN).putAlias(AliasMetaData.builder("aaa_alias1")))
+                    .put(indexBuilder("bbb").state(State.OPEN).putAlias(AliasMetaData.builder("bbb_alias1")))
+                    .put(indexBuilder("ccc").state(State.CLOSE).putAlias(AliasMetaData.builder("ccc_alias1")))
+                    .build();
+            if (indicesOptions.expandWildcardsOpen() || indicesOptions.expandWildcardsClosed() || indicesOptions.allowNoIndices()) {
+                String[] concreteIndices = metadata.concreteIndices(indicesOptions, allIndices);
+                assertThat(concreteIndices, notNullValue());
+                int expectedNumberOfIndices = 0;
+                if (indicesOptions.expandWildcardsOpen()) {
+                    expectedNumberOfIndices += 2;
+                }
+                if (indicesOptions.expandWildcardsClosed()) {
+                    expectedNumberOfIndices += 1;
+                }
+                assertThat(concreteIndices.length, equalTo(expectedNumberOfIndices));
+            } else {
+                try {
+                    metadata.concreteIndices(indicesOptions, allIndices);
+                    fail("calling concreteIndices for _all, with no wildcard expansion and allowNoIndices=false should throw exception");
+                } catch (ElasticsearchIllegalArgumentException | IndexMissingException e) {
+                    // expected exception
+                }
+            }
+        }
+    }
+
+    /**
+     * test resolving wildcard pattern that matches no index of alias for random IndicesOptions
+     */
+    @Test
+    public void concreteIndicesWildcardEmptyRandom() {
```

Comment:
I'll change that for all the tests (also the existing ones) in MetaDataTests then.